### PR TITLE
Next

### PR DESCRIPTION
**Pull request by jxnblk**
Install with

```sh
npm i @compositor/x0@next
```

## API Changes
- Supports nested routes
- Adds support for `.md` files, using MDX
- Configurable default styles for markdown and MDX files
- Adds LiveEditor and LivePreview capabilities to markdown
- Adds `--match` option for filtering files
- Adds `ignore` option for pages
- Optional SidebarLayout component
- Optional Library component
- Support for 404 pages

## Internal Changes
- Uses mdx loader for md/mdx/jsx files

## Breaking Changes
- Specifying scope in JSX front matter is no longer supported
- Removes support for glamor and glamorous static CSS extraction
- Previously undocumented `app` option is no longer used

### CODE REVIEW

File: lib/build.js [hunk 1]
```diff
@@ -88,35 +82,25 @@ const renderHTML = ({
       const { ServerStyleSheet } = require('styled-components')
       const sheet = new ServerStyleSheet()
       html = render(
-        sheet.collectStyles(
-          React.createElement(App.default, { routes, path })
```
Comment (jxnblk) [inline lib/build.js]: Need to make sure the `createElement` function isn't needed inside `sheet.collectStyles`

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 
 # x0
 
-Zero-config React development environment and static site generator
+Document & develop React components without breaking a sweat
```
Comment (johno) [inline README.md]: 😅

File: cli.js [hunk 3]
```diff
@@ -28,11 +28,13 @@ const cli = meow(`
   ${chalk.gray('Options')}
 
       --webpack       Path to webpack config file
+      --match         String to match routes against using minimatch
 
     ${chalk.gray('Dev Server')}
 
       -o --open       Open dev server in default browser
       -p --port       Port for dev server
+      --analyze       Runs with webpack-bundle-analyzer plugin
```
Comment (johno) [inline cli.js]: 😎

File: lib/mdx-fm-loader.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// front-matter loader for mdx
+const matter = require('gray-matter')
+const stringifyObject = require('stringify-object')
+
+module.exports = async function (src) {
+  const callback = this.async()
+  const { content, data } = matter(src)
+
+  const code = `export const frontMatter = ${stringifyObject(data)}
```
Comment (johno) [inline lib/mdx-fm-loader.js]: Slick.

**Pull request merged**